lib/logster/message.rb

```diff
@@ -203,6 +203,23 @@ def =~(pattern)
       end
     end
 
+    def self.scrub_params(params)
+      if Array === params
+        params.map! { |p| scrub_params(p) }
```

Comment:
slight concern here is that this is amending input if it is invalid as a side effect. I worry that all this work is not really logsters concern. 

Plus, maybe it should just do this work when you click on actual messages, that way we don't risk fiddling with inputs?

Reply:
We can do this when retrieving log messages and it would solve the problem for rails apps, but for non-rails apps that integrate logster, they would probably need the scrubbing to be done when the log message is received, otherwise I think the log message would be completely lost. This is because when a message is received we call `.to_json` on the message env to save it in redis, but `.to_json` fails if it's called on a hash that contains strings with invalid encoding. ActiveSupport on the other hand, has a monkeypatch for `.to_json` which makes it not fail even if the hash has strings with invalid encoding (I don't think that's the primary reason for the patch).

I can change this so it doesn't amend input, would you still be comfortable with this PR?

Reply:
yeah, this is a pickle... I guess we add a note to the README warning consumers not to shove invalid encoding into logster for rack apps? 

I say amend it so it does not fiddle with inputs for now cause if fixes rails which is the biggest consumer here.

Reply:
@SamSaffron I've updated this so the scrubbing is done when retrieving messages and added a note in README about non-rails apps. Let me know what you think.